### PR TITLE
Migrate to Spring Boot 3, java 17

### PR DESCRIPTION
**Pull request by aiecel**
Super fresh, super clean

### CODE REVIEW

File: crap-advisor-api/build.gradle.kts [hunk 1]
```diff
@@ -17,48 +16,54 @@ kapt {
     }
 }
 
+noArg {
+    annotations(
+        "jakarta.persistence.Entity",
+        "jakarta.persistence.MappedSuperclass",
+        "jakarta.persistence.Embeddable"
```
Comment (akichula) [inline crap-advisor-api/build.gradle.kts:23]: Explain this one.

File: crap-advisor-api/src/main/kotlin/com/aiecel/crapadvisor/api/controller/RestroomController.kt [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
     @GetMapping
     @Operation(summary = "Get all restrooms")
-    fun getAll() = service.getAll().map { restroomMapper.map(it) }
+    fun getAll() = service.getAll().map { mapper.map(it) }
 
     @PostMapping
     @Operation(summary = "Save a new restroom")
     fun save(@RequestBody @Validated request: AddRestroomRequest) =
-        restroomMapper.map(
+        mapper.map(
```
Comment (akichula) [inline crap-advisor-api/src/main/kotlin/com/aiecel/crapadvisor/api/controller/RestroomController.kt:30]: Glad you did it. Good

File: crap-advisor-api/build.gradle.kts (hunk 1 above)
Comment (aiecel) [reply crap-advisor-api/build.gradle.kts:23]: hibernate needs entities to have an empty default constructor, in java we can use lombok, in kotlin we can use this plugin that generates constructors automatically for classes marked with those annotations

dizzy technique man

File: crap-advisor-api/src/main/kotlin/com/aiecel/crapadvisor/api/controller/RestroomController.kt (hunk 2 above)
Comment (aiecel) [reply crap-advisor-api/src/main/kotlin/com/aiecel/crapadvisor/api/controller/RestroomController.kt:30]: one small step for a file, one giant leap for a whole project

**Review by akichula — APPROVED** [review]
Now it's can be merged successfully

**Pull request merged**